app/components/Swapbox.tsx

```diff
@@ -2,11 +2,117 @@
 
 import { Button, IconButton } from "swapr-ui";
 import { SwapInput } from "./ui/SwapInput";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { erc20Abi, formatEther, parseEther, Address } from "viem";
+import { useAccount, useConfig, useReadContract } from "wagmi";
+import { waitForTransactionReceipt, writeContract } from "wagmi/actions";
+import MarketABI from "@/abi/market.json";
+
+const WXDAI_ADDRESS = "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d";
+
+// Removes the given fraction from the given integer-bounded amount and returns the value as an original type.
+const removeFraction = (amount: bigint, fraction: number): bigint => {
+  if (fraction >= 1 || fraction <= 0)
+    throw `The given basisPoints ${fraction} is not in the range [0, 1].`;
+
+  const basisPoints = 10000;
+
+  const fractionInBasisPoints = fraction * basisPoints;
+
+  const keepFraction = basisPoints - fractionInBasisPoints;
+
+  return (amount * BigInt(keepFraction)) / BigInt(basisPoints);
+};
+
+export const Swapbox = ({ id }: { id: Address }) => {
+  const { address } = useAccount();
+  const config = useConfig();
 
-export const Swapbox = () => {
   const [tokenInAmount, setTokenInAmount] = useState("");
   const [tokenOutAmount, setTokenOutAmount] = useState("");
+  const [minAmountOut, setMinAmountOut] = useState<bigint>();
+
+  const outcomeIndex = 0;
+  const slippage = 0.01;
```

Comment:
this should be a constant right? All caps.

Reply:
We can have that. But since it's going to be a variable value I used lowercase. Will update `slippage` but `outcomeIndex` will be a variable value soon.